src/waveform/renderers/waveformrendermark.cpp

```diff
@@ -123,7 +123,7 @@ void WaveformRenderMark::slotCuesUpdated() {
         // because WaveformMarkSet stores one mark for each hotcue.
         WaveformMarkPointer pMark = m_marks.getHotCueMark(hotCue);
         if (pMark.isNull()) {
-        	continue;
+            continue;
         }
 
         WaveformMarkProperties markProperties = pMark->getProperties();
```

Comment:
This pattern of writing properties becomes more and more bad, as the content of WaveformMarkProperties grows. 
We may add helper functions to WaveformMark to set the label and color and make markProperties a const & here. 
Aternatively use a pMark->getProperties() overload returning a non-const raw pointer (the caller promises to not store raw pointer, which is the case here) to manipulate the WaveformMarkProperties directly.

Reply:
I'm not clear what the role of WaveformMark versus WaveformMarkProperties are intended to be.

Reply:
I don't know it either. So you may merge it to WaveformMark, if you like.

Reply:
I merged WaveformMarkProperties into WaveformMark. This makes the code simpler IMO.